.ado/jobs/universal.yml

```diff
@@ -1,147 +1,227 @@
-    strategy:
-      matrix: # Why we only build some flavors: https://github.com/microsoft/react-native-windows/issues/4308
-        # Start Continuous only
-        ${{ if eq(parameters.buildEnvironment, 'Continuous') }}:
-          ArmDebug:
-            BuildConfiguration: Debug
-            BuildPlatform: ARM
-        # End Continuous only
-        ArmRelease:
-          BuildConfiguration: Release
-          BuildPlatform: ARM
-        # Start Continuous only
-        ${{ if eq(parameters.buildEnvironment, 'Continuous') }}:
-          Arm64Debug:
-            BuildConfiguration: Debug
-            BuildPlatform: ARM64
-          Arm64Release:
-            BuildConfiguration: Release
-            BuildPlatform: ARM64
-          X64Debug:
-            BuildConfiguration: Debug
-            BuildPlatform: x64
-        # End Continuous only
-        X64Release:
-          BuildConfiguration: Release
-          BuildPlatform: x64
-        X86Debug:
-          BuildConfiguration: Debug
-          BuildPlatform: x86
-          LayoutHeaders: true
-        # Start Continuous only
-        ${{ if eq(parameters.buildEnvironment, 'Continuous') }}:
-          X86Release:
-            BuildConfiguration: Release
-            BuildPlatform: x86
-        # End Continuous only
-    pool:
-      vmImage: $(VmImage)
-    timeoutInMinutes: 60
-    cancelTimeoutInMinutes: 5
+  parameters:
+    - name: buildEnvironment
+      type: string
+      default : PullRequest
+      values: 
+      - PullRequest 
+      - Continuous 
+    - name: buildMatrix
+      type: object
+      default:
+        - BuildEnvironment: PullRequest
+          Matrix:
+            - BuildConfiguration: Release
+              BuildPlatform: ARM
+            - BuildConfiguration: Release
+              BuildPlatform: x64
+            - BuildConfiguration: Debug
+              BuildPlatform: x86
+              LayoutHeaders: true
+        - BuildEnvironment: Continuous
+          Matrix:
+            - BuildConfiguration: Debug
+              BuildPlatform: ARM
+            - BuildConfiguration: Release
+              BuildPlatform: ARM
+            - BuildConfiguration: Debug
+              BuildPlatform: ARM64
+            - BuildConfiguration: Release
+              BuildPlatform: ARM64
+            - BuildConfiguration: Debug
+              BuildPlatform: x64
+            - BuildConfiguration: Release
+              BuildPlatform: x64
+            - BuildConfiguration: Debug
+              BuildPlatform: x86
+              LayoutHeaders: true
+            - BuildConfiguration: Release
+              BuildPlatform: x86
+      
+  jobs:
+    - ${{ each config in parameters.buildMatrix }}:
+      - ${{ if eq(config.BuildEnvironment, parameters.buildEnvironment) }}:
+        - ${{ each matrix in config.Matrix }}:
+            - job: UniversalBuild${{ matrix.BuildPlatform }}${{ matrix.BuildConfiguration }}
```

Comment:
Nit that I noticed for how these are displayed next to other jobs, we have a mix of lowercase and uppercase `x` and `x`

Reply:
Thanks for pointing out. I beleive I have addressed most of these in the next update.